src/main/kotlin/com/devfloor/untitled/article/application/ArticleRequest.kt

```diff
@@ -0,0 +1,19 @@
+package com.devfloor.untitled.article.application
+
+import com.devfloor.untitled.article.domain.Article
+import com.devfloor.untitled.user.domain.User
+
+
+data class ArticleRequest(
+    val options: List<String>,
+    val title: String? = null,
+    val anonymous: Boolean,
+    val content: String,
+    val hashtags: List<String>
+) {
+    fun toArticleWithUser(articleRequest: ArticleRequest, user: User): Article {
+        return Article(articleRequest.title, articleRequest.anonymous, articleRequest.content, user)
+    }
```

Comment:
1. DTO -> Entity로 변경해주는 코드는 Article에 두는게 어떨까요 !?
user와 articleDTO를 사용해서 Article 객체를 만들어주는 것이기 때문에 Article에 있는게 좋을 것 같아요 !

2. 변수명도 `of` 혹은 `toEntity` 바꿔서, 저 메서드를 사용하는 부분(Facade)에서 
`Article.of(articleRequest, user)` 이와같이 사용하도록 한다면, 가독성 측면에서도 좋아질 것 같아요 !

```suggestion
    companion object {
        fun of(
            articleRequest: ArticleRequest,
            user: User
        ) = Article(
            title = articleRequest.title,
            anonymous = articleRequest.anonymous,
            content = articleRequest.content,
            author = user
        )
    }
```

Reply:
어, 이 부분은 전 지금이 맞다고 생각합니다!

article 엔티티 내부에서 dto를 article로 만들면 article이 articleRequest를 의존하게 됩니다. 의존 방향이 역전돼요!
만약 프론트에서 request 형태가 바뀌면 변경 범위가 articleRequest까지 영향을 미쳐야지 도메인 로직인 article 엔티티까지 미치면 안됩니다!

그래서 지금처럼 DTO에서 Entity를 만들어주는게 맞는 것 같네요!

Reply:
위처럼 로직이 그대로 Request에 존재할 경우 주원이형이 2번에서 제안한 것처럼 로직을 바꿀 필요는 없을 것 같습니다.
추가적으로 여기서는 적용하는게 옳지 않아보이지만 주원이형이 제안한 방법을 정적 팩토리 메서드라고 부릅니다.

이건 아무 글이나 링크 걸긴 했는데 [생성자 대신 정적 팩토리 메서드를 고려하라](https://velog.io/@litien/%EC%83%9D%EC%84%B1%EC%9E%90-%EB%8C%80%EC%8B%A0-%EC%A0%95%EC%A0%81-%ED%8C%A9%ED%86%A0%EB%A6%AC-%EB%A9%94%EC%84%9C%EB%93%9C%EB%A5%BC-%EA%B3%A0%EB%A0%A4%ED%95%98%EB%9D%BC) 이런 종류의 글을 읽어보면 좀 더 도움이 되지 않을까 싶습니다.

Reply:
아.. 의존성을 생각해봤을땐 DTO에서 만드는게 맞을거같네요 !
생각해보니까 도메인에서 dto를 입력를 받아 무언가를 만들어주는건 오히려 이상해보일수있겠네요 !
그렇다면 이 부분만 반영해주셔도 좋을거같아요 !
```suggestion
    fun toArticleWithUser(articleRequest: ArticleRequest, user: User): Article {
        return Article(
            title = articleRequest.title,
            anonymous = articleRequest.anonymous,
            content = articleRequest.content
            author = user)
    }
```

Reply:
이 부분 보면서 DTO와 entity를 맵핑해주는 좋은방법이 뭘까 고민중이긴 한데요, 일단 코틀린에서는 직접 맵핑해줄 때 각각이 어디에 맵핑될지 가독성자체가 좋아서 많이들 그냥 맵핑하기도 하더라구요 !

아 그리고 찾아보다가 ModelMapper? 라는애를 우연히 보게 되었는데 Entity to dto, dto to Entity 잘 할수있게 도와주는 것 같더라구요. (그런데 이걸 사용하는게 적절할지는 사실 잘 모르겠어요 아직 저도 찾아보고있어서..)

그래서 이 부분도 한번 참고해보시면 좋을 것 같습니다.

Reply:
ModelMapper에 대해서 조금 찾아봤는데 일단 Kotlin 레퍼런스가 많이 없고 성능 이슈가 있는 것 같습니다.
따라서 Kotlin에서는 직접 매핑하는게 제일 간단하고 좋아보이네요 !

Reply:
변경완료했습니다!